apps/web/src/app/[locale]/challenge/[slug]/solutions/[solutionId]/getSolutionIdRouteData.ts

```diff
@@ -4,11 +4,15 @@ import { type Session } from '@repo/auth/server';
 import { redirect } from 'next/navigation';
 
 export const getSolutionIdRouteData = cache(
-  async (challengeId: string, solutionId: string, session: Session | null) => {
+  async (slug: string, solutionId: string, session: Session | null) => {
+    const challenge = await prisma.challenge.findFirstOrThrow({
```

Comment:
we used to be able to get id from the url but since its slug i need to query and get the id.

i could make slug a foreign key to solutions but unsure if thats a good idea.. id seems the only reasonable FK here